src/commands/positive.ts

```diff
@@ -72,17 +75,53 @@ export default class Positive extends ApiKeyCommand {
     }
 
     this.schema = new OASSchema(oasSchemaOptions);
+
     await this.buildOperationExamples();
 
-    this.validateParameters();
+    await Promise.all(
+      this.operationExamples.map(async (operationExample) => {
+        let failures: ValidationFailure[] = [];
+        let warnings: ValidationWarning[] = [];
+        const { operation, exampleGroup } = operationExample;
+        const parameterSchemaValidator = new ParameterSchemaValidator(
+          operation,
+        );
+        parameterSchemaValidator.validate();
 
-    const responses = await Promise.all(this.executeRequests());
+        failures = [...failures, ...parameterSchemaValidator.failures];
+        warnings = [...warnings, ...parameterSchemaValidator.warnings];
 
-    const mergedResponses = responses.reduce((merged, response) => {
-      return Object.assign(merged, response);
-    }, {});
+        const parameterValidator = new ParameterValidator(exampleGroup);
+        parameterValidator.validate();
```

Comment:
If a parameter fails the parameter schema validator, does it make sense to still validate it?

It looks like the current behavior for if a parameter has both schema and content set for example, is that the ParameterValidator will try to validate its schema when it gets to the example check

Reply:
Yeah, i wasn't super sure about this myself. We could potentially try to mark the parameter itself as failed, and remove it from all the example groups, but the `parameterValidator` really validates example groups, which are combinations of parameters. I figured that we'd still want to give them any information on the example groups validation though, so it didn't seem right to skip it entirely if the parameter schema validator failed.

This talk has led me to believe that i should rename ParamterValidator to ExampleGroupValidator. Thoughts?

Reply:
That sounds good to me! Agree it seems valuable to still run the example group validator because there may be other errors that could be fixed. We could always wait and see if tracking validation status per parameter would be valuable if this is too noisy, but I also don't think these issues with schema problems will be very common so I don't imagine it'll be very noisy